src/cogent3/app/composable.py

```diff
@@ -992,6 +992,8 @@ def _call(self, val, *args, **kwargs):
     # todo we should get the source information from val here
     if self.app_type is not LOADER and self.input:  # passing to connected app
         val = self.input(val, *args, **kwargs)
+        if isinstance(val, NotCompleted):
+            return val
```

Comment:
Here, Do we need to check `skip_not_completed`?

Reply:
Good point!